OpenSim/Common/Component.h

```diff
@@ -2285,70 +2294,50 @@ OpenSim_DECLARE_ABSTRACT_OBJECT(Component, Object);
 #endif
 
     template<class C>
-    const C* traversePathToComponent(const std::string& path) const
+    const C* traversePathToComponent(ComponentPath path) const
     {
+        // Get rid of all the ".."'s that are not at the front of the path.
+        path.cleanPath();
+
```

Comment:
This method takes a ComponentPath (passed by value) and modifies it. Would user expect path to change here? Can the method take const reference, make a local copy to change so that a user of this function doesn't see unexpected side-effects? or am I misreading the implementation.

Reply:
@aymanhab making a copy of a const reference achieves the same thing here as pass by value. The user's path is not modified.

Reply:
Good point @chrisdembia though the signature would be different/more explicit about it.